services/asymkey/ssh_key.go

```diff
@@ -41,8 +43,8 @@ func DeletePublicKey(doer *user_model.User, id int64) (err error) {
 	committer.Close()
```

Comment:
@delvh like this function

Reply:
Ah, so you mean an explicit `Close` on top of the deferred `Close`?

Reply:
Why? deferd is alreadycalled when the scope is left.

Reply:
I think @lunny's point is that it should also be called before it is left so that the transaction isn't present for too long.

Reply:
So that the transaction close before invoking `SyncXXX` functions.

Reply:
So, is this one thing blocking the merge, or can we merge this PR?

Reply:
> So, is this one thing blocking the merge, or can we merge this PR?

I would not say it's a blocker.